rosbag2_cpp/include/rosbag2_cpp/writer.hpp

```diff
@@ -74,6 +74,12 @@ class ROSBAG2_CPP_PUBLIC Writer
    * more storage and converter options.
    *
    * \param storage_uri URI of the storage to open.
+   * \throws runtime_error if 
+   * database directory already exists,
+   * failed to create database directory,
+   * no storage could be initialized,
+   * invalid bag splitting size given,
+   * max cache size less or equal 0 when snapshot mode is enabled.
```

Comment:
I think exception behavior is dependent on the writer implementation, so how about the following?

```suggestion
   * \throws runtime_error if internal error happens.
```